orc8r/cloud/go/services/certifier/analytics/calculations/lifespan.go

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2020 The Magma Authors.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package calculations
+
+import (
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"io/ioutil"
+	"math"
+	"strings"
+	"time"
+
+	"magma/orc8r/cloud/go/services/analytics/calculations"
+	"magma/orc8r/cloud/go/services/analytics/protos"
+	"magma/orc8r/cloud/go/services/analytics/query_api"
+	"magma/orc8r/lib/go/metrics"
+
+	"github.com/golang/glog"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type CertLifespanCalculation struct {
+	calculations.BaseCalculation
+}
+
+const (
+	DefaultCertsDirectory = "/var/opt/magma/certs/"
+	RootCert              = "rootCA.pem"
+	AdminCert             = "admin_operator.pem"
+	ControllerCert        = "controller.crt"
+	CertifierCert         = "certifier.pem"
+	FluentdCert           = "fluentd.pem"
+)
+
+func (x *CertLifespanCalculation) Calculate(prometheusClient query_api.PrometheusAPI) ([]*protos.CalculationResult, error) {
+	glog.V(2).Infof("Calculating %s", metrics.CertLifeHoursMetric)
+
+	var results []*protos.CalculationResult
+
+	metricConfig, ok := x.AnalyticsConfig.Metrics[metrics.CertLifeHoursMetric]
+	if !ok {
+		glog.Errorf("%s metric not found in metric config", metrics.CertLifeHoursMetric)
+		return results, nil
+	}
+
+	var certs = [...]string{RootCert, AdminCert, ControllerCert, CertifierCert, FluentdCert}
```

Comment:
No need for an array literal here, just use a slice
```suggestion
	var certs = []string{RootCert, AdminCert, ControllerCert, CertifierCert, FluentdCert}
```